src/api/sudo_network_unstable_watch.md

```diff
@@ -0,0 +1,129 @@
+
+`direction` indicates whether the connection was initiated locally (`out`) or by the remote (`in`). The value in the `direction` field must always be the same for all the events related to a specific connection. In other words, a connection never changes direction during its lifetime.
+
+`when` is an integer containing the UNIX timestamp in milliseconds, in other words the number of milliseconds since the UNIX epoch ignoring leap seconds.
+
+### substreamState
+
+```json
+{
+    "event": "substreamState",
+    "connectionId": "...",
+    "substreamId": "...",
+    "status": "open" | "closed",
+    "protocolName": "...",
+    "direction": "in" | "out",
+    "when": ...
+}
+```
+
+A `substreamState` event is generated when a new connection attempt is started, when a connection has finished its handshaking phase, or when a connection is terminated.
+
+`connectionId` is an opaque string representing this specific connection. It must always correspond to a connection whose latest `status` is `open`.
+
+`substreamId` is an opaque string representing this specific substream within the connection. Each substream is identified by the `connectionId` + `substreamId` tuple rather than just the `substreamId` alone. The JSON-RPC server is allowed to use the same value of `substreamId` for two different substreams belonging to two different connections.
+
+`status` indicates the state of the substream: `open` if the substream is "alive", or `closed` if the substream is dead. A substream is considered "alive" if the JSON-RPC server allocates resources for this substream, even if the remote isn't aware of said substream.
+
+Each `substreamState` event where `status` equal to `closed` must follow a previous `substreamState` even for that same substream where `status` was `open`. In other words, the JSON-RPC server is not allowed to omit event the `open` event.
+
+Once a `substreamState` event with `status` equal to `closed` is generated, the `substreamId` is unallocated. Any further usage of the same `substreamId` in the context of that `connectionId` designates a different substream instead.
+
+`protocolName` is a string indicating the multistream-select protocol name that was negotiated.
+
+`direction` indicates whether the substream was initiated locally (`out`) or by the remote (`in`). Note that this is not the same thing as the `direction` of the corresponding connection. The value in the `direction` field must always be the same for all the events related to a specific substream. In other words, a substream never changes direction during its lifetime.
+
+`when` is an integer containing the UNIX timestamp in milliseconds, in other words the number of milliseconds since the UNIX epoch ignoring leap seconds.
+
+### missedEvents
+
+```json
+{
+    "event": "missedEvents"
+}
+```
+
```

Comment:
Would it also be beneficial to have an optional even reported for cases of miss-behaving / reputation banning?

Reply:
The networking in general has a shit-ton of concepts. Beyond reputations, you could also think of reporting ping times, handshake and request sizes, k-buckets states, grandpa neighbor packets, block announces, etc.

This PR focuses on just open connections and substreams because 1) these concepts are easy to understand 2) we know for sure that will never change in Polkadot.